tests/test_dataset.py

```diff
@@ -1054,6 +1056,39 @@ def test_get_private_item(self):
         item = ds.get_private_item(0x0009, 0x02, "Creator 2.0")
         assert 2 == item.value
 
+    def test_add_unknown_private_tag(self):
+        ds = Dataset()
+        with pytest.raises(ValueError, match="Tag must be private"):
+            ds.add_private("Creator 1.0", 0x0010, 0x01, "VALUE", VR.SH)
+        with pytest.raises(ValueError, match="Private creator must have a value"):
+            ds.add_private("", 0x0011, 0x01, "VALUE", VR.SH)
+        with pytest.raises(
+            KeyError,
+            match="Private creator 'Creator 1.0' not in the private dictionary",
+        ):
+            ds.add_private("Creator 1.0", 0x0011, 0x01, "VALUE")
```

Comment:
Is it possible, or wise, to relax this restriction that it is an existing private creator from the dictionary?  I wonder whether people finding this convenience method might just assume it can work in either case.  Could the method have additional parameters to allow it to create the creator, basically making a call to the `add_private_dict_entry` as done before the method call in one of the tests below?  I'm not clear on how awkward the extra parameters might be.

Reply:
> Hmm.. I like add_new_private as a nice parallel to the existing add_new.

Ok, I'll go with that then.

>  Could the method have additional parameters to allow it to create the creator

It does create it, just does not register it.   
In principle, we could always call `add_private_dict_entry` if it is not known yet - I'm not sure about this, though... Registering the private tag is helpful for reading the tag, and normally this would be done in a separate process, so it does not help much if we register the tag during writing if it is not read back in the same process.
Also, as you wrote, we would need to add additional (optional) arguments, e.g. the tag description and the VM, and I don't like this too much.

>  I wonder whether people finding this convenience method might just assume it can work in either case

If the VM is not given and the tag is not registered, it cannot work, of course. I could make the `vr` argument mandatory of course - in this case it would just be reduntant (same as in `PrivateBlock.add_new`). I also have thought about making the VR argument in `PrivateBlock.add_new` optional, but that would mean the argument order has to be changed, so I didn't do it.

Reply:
On further thought (and with your reasons), I think the way you have done it here is the best balance of ease of use and non-complicated parameters, etc.  I'll merge...